packages/teuchos/parameterlist/src/Teuchos_YamlParser.cpp

```diff
@@ -659,17 +662,11 @@ void generalWriteString(const std::string& str, std::ostream& yaml)
 
 void generalWriteDouble(double d, std::ostream& yaml)
 {
-  yaml << std::showpoint << std::setprecision(8);
-  if(d < 1e6 && d > 1e-5)
-  {
-    //use regular notation
-    yaml << d;
-  }
-  else
-  {
-    yaml << std::scientific << d;
-  }
-  yaml << std::fixed;
+  std::ios saved_state(0);
+  saved_state.copyfmt(yaml);
+  yaml << std::noshowpoint << std::scientific << std::setprecision(15);
```

Comment:
It think if you want full precision, you need to write out **at least** 17 digits to ensure that you read back in the same floating point number.   See:

* https://en.wikipedia.org/wiki/Double-precision_floating-point_format

That is critical if you want a binary restart.